docs/jep/12049-type-annotations.md

```diff
@@ -197,6 +197,40 @@ We will revisit this question in the future once support for such features stabi
 
 Similarly, regarding [jaxtyping](http://github.com/google/jaxtyping) project, despite its promise as a project, at this point we consider its goals beyond the scope of the simpler level of type annotations we would like for the jax core project. This is a decision we could revisit at a future date.
 
+### Design Considerations
+
+#### Following NumPy's Lead
+
+One possible model for JAX's type annotations is those provided by the [`numpy.typing`](https://numpy.org/devdocs/reference/typing.html) module.
+An advantage of this approach is that it is that the API is already familiar to many; most notably `numpy.typing` provides:
+
+- `NDArray`, which is a generic array type used only for type annotations
+- `np.ndarray`, which is the array type used for instance checks, and also works for type annotations. In Python 3.9, can use constructs of the form `np.ndarray[shape, dtype]`, but this is currently poorly documented and it's unclear how well supported that is at the moment.
+
+JAX currently implements {class}`jax.numpy.ndarray` for use with runtime `isinstance` checks. It uses a metaclass override to ensure that tracers also will pass `isinstance(tracer, jnp.ndarray)`, but it does not currently have any mechanism to ensure that tracers will be valid for `jnp.ndarray` annotations. If we were to follow numpy's approach to type annotation, we could do something like the following:
+- add `jax.typing.NDArray` for use with type annotations
+- add `TYPE_CHECKING` logic to ensure that `jnp.ndarray` can be used as an annotation for tracers; and possibly add class-level `__getitem__` to match numpy's shape and dtype specification.
```

Comment:
Are these two bullet points meant as mutually exclusive possibilities? (Since having both sounds redundant.)

Reply:
They are meant to outline two aspects of following numpy's precedent.